### PR TITLE
Implement the secrets clientside

### PR DESCRIPTION
**Pull request by nov1n**
This pull request implements the secrets API on the client side. It supports the four commands:

- nerd secret list
- nerd secret create
- nerd secret delete
- nerd secret describe

Three of these commands are self explanatory and function exactly the same as commands for other resources. The create command deserves some extra attention.

### nerd secret create
We distinguish two types of secrets that are currently saved on the server in the same data structure, `Secret`. 

#### Opaque

The first type of secret is the *Opaque* secret, stemming from the name it has in kubernetes. This secret can be mounted by a user as environment variables. The opaque secret can be created with (--type=opaque is optional as this is the default):
```
nerd secret create my-secret key=val --type=opaque
```

#### Registry

The second type is the *Registry* secret (pull secret in kubernetes). This secret is used by the init container (cuteur) to pull a private image, nerdalize it, and run it. The registry secret can be created with:
```
nerd secret create repo.treescale.com key=value --username=nov1n --password=123123123 --type=registry
```
The name of the secret has to match the location of the registry. The default dockerhub registry is index.docker.io.

### TODO
There are a few things I did not have time for, these are listed below:

- [ ] Add proper helper strings (--help) to the secret subcommands.
- [ ] Test edge cases, an example of this is `nerd secret create key=val --type=registry --username=foo --password=bar` works fine, although a user should either provide a key,value pair *or* a username and password.
- [ ] Do proper validation, e.g. number of arguments supplied, type of the arguments, etc. Kubernetes does some checking for us as the server proxies the request to kubernetes and returns errors if any.

### CODE REVIEW

File: command/secret.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	*command
+}
+
+// SecretFactory returns a factory method for the join command
+func SecretFactory() (cli.Command, error) {
+	comm, err := newCommand("nerd secret <subcommand> {opaque, registry}", "set and list secrets", "", nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create command")
+	}
+	cmd := &Project{
```
Comment (LilliJane) [inline command/secret.go]: Shouldn't we use the Secret struct isn't of the Project one?

Comment (nov1n) [inline command/secret.go]: Good catch! That is a mistake indeed, I'll try to write up some documentation about the PR to make your life easier (:

**Comment by nov1n** [comment]
@dhrp @LilliJane I tried to write up what I thought was useful about the PR, I will be at Nerdalize on the 21th of July for any questions left. I hope you manage to pick it up, good luck!
